recipes-extended/pam/libpam_1.%.bbappend

```diff
@@ -1 +1,16 @@
 FILESEXTRAPATHS:prepend := "${THISDIR}/${BPN}:"
+
+RCONFLICTS:pam-plugin-faillock += "ni-auth"
+
+pkg_postinst:pam-plugin-faillock:append() {
+	# configure faillock settings
+	sed -i 's/^# audit$/audit/' "${sysconfdir}/security/faillock.conf"
+	sed -i 's/^# silent$/silent/' "${sysconfdir}/security/faillock.conf"
+	sed -i 's/^# deny = 3$/deny = 3/' "${sysconfdir}/security/faillock.conf"
+	sed -i 's/^# fail_interval = 900$/fail_interval = 900/' "${sysconfdir}/security/faillock.conf"
+	sed -i 's/^# unlock_time = 600$/unlock_time = 0/' "${sysconfdir}/security/faillock.conf"
+
+	# enable faillock
+	sed -E -i 's/^(.+)success=1(.+)$/auth    requisite pam_faillock.so preauth\n\1success=2\2\nauth    [default=die] pam_faillock.so authfail/' "${sysconfdir}/pam.d/common-auth"
+	echo "auth    sufficient pam_faillock.so authsucc" >> "${sysconfdir}/pam.d/common-auth"
```

Comment:
I could not find a good use of `include`. If you `include` a missing file, then it fails the authentication. I tried several different syntaxes, and I was not able to `include` a directory or wildcard.

Reply:
:( Well, in that case let's just make sure that you can also remove this package and have the rule get correctly removed from the `common-auth` file.

You may want to match on `.*pam_faillock.so.*` for the `sed` regex. None of the other rules should refer to that library.

Reply:
Can we not replace the whole `common-auth` with our version like meta-cloud-service's [bbappends](https://github.com/ni/meta-cloud-services/blob/nilrt/master/next/meta-openstack/recipes-extended/libpam/libpam_openstack.inc) does instead of using `sed`? Might be cleaner that way.

> Since we're plugging PAM modules from packages

Do we expect to add more rules to this file from packages other than `libpam-*`?

Reply:
The `libpam-runtime` package owns `common-auth`. The `meta-cloud-services` example you mentioned is updating the `libpam-runtime` package. This PR is changing the `pam-plugin-faillock` package, which does not own `common-auth`.